src/rp2_common/pico_status_led/include/pico/status_led.h

```diff
@@ -0,0 +1,121 @@
+#if !defined PICO_STATUS_LED_PIN && defined PICO_DEFAULT_LED_PIN
+#define PICO_STATUS_LED_PIN PICO_DEFAULT_LED_PIN
+#endif
+
+// PICO_CONFIG: PICO_STATUS_LED_WS2812_PIN, Change the pin used for the ws2812 status led from the value configured in the board headers, type=int, group=pico_status_led
+#if !defined PICO_STATUS_LED_WS2812_PIN && defined PICO_DEFAULT_WS2812_PIN
+#define PICO_STATUS_LED_WS2812_PIN PICO_DEFAULT_WS2812_PIN
+#endif
+
+// PICO_CONFIG: PICO_STATUS_LED_USE_WS2812_PIN, If a ws2812 status pin is configured, use this instead of a normal status led , type=int, group=pico_status_led
+#if PICO_STATUS_LED_USE_WS2812_PIN && defined PICO_STATUS_LED_WS2812_PIN
+#undef PICO_STATUS_LED_PIN
+#endif
+
+/*! \brief Initialise the status led
+ *  \ingroup pico_status_led
+ *
+ *  Initialise the status led before use.
+ *
+ * \return Returns PICO_OK if the led was initialised, otherwise an error is returned
+ */
+int pico_status_led_init(void);
+
+#if defined PICO_STATUS_LED_WS2812_PIN
+
+/*! \brief Set the ws2812 status led on or off
+ *  \ingroup pico_status_led
+ *
+ * \param led_on True to turn the led on. Pass False to turn the led off
+ */
+void pico_status_led_ws2812_set(bool led_on);
+
+/*! \brief Set the ws2812 status led pixel value when on or off
+ *  \ingroup pico_status_led
+ *
+ * The ws2812 status led defaults to white for on and blank for off. Use this function to change the default RGB values.
+ *
+ * \param led_on True to set the pixel value when the led is on. Pass False to set the pixel value when the led is off.
+ */
+void pico_status_led_ws2812_set_pixel(bool led_on, uint8_t r, uint8_t g, uint8_t b);
+
+/*! \brief Get the state of the ws2812 status led
+ *  \ingroup pico_status_led
+ *
+ * \return True if the led is on, or False if the led is off
+ */
+bool pico_status_led_ws2812_get(void);
+
+#endif // PICO_STATUS_LED_WS2812_PIN
+
+/*! \brief Set the status led on or off
+ *  \ingroup pico_status_led
+ *
+ * \param led_on True to turn the led on. Pass False to turn the led off
+ */
+static inline void pico_status_led_set(bool led_on) {
+#if defined PICO_STATUS_LED_PIN
+    gpio_put(PICO_STATUS_LED_PIN, led_on);
+#elif defined PICO_STATUS_LED_WS2812_PIN
+    pico_status_led_ws2812_set(led_on);
+#elif defined CYW43_WL_GPIO_LED_PIN
+    // Ask the wifi "driver" to set the GPIO on or off
+    cyw43_arch_gpio_put(CYW43_WL_GPIO_LED_PIN, led_on);
+#endif
+}
+
+/*! \brief Get the state of the status led
+ *  \ingroup pico_status_led
+ *
+ * \return True if the led is on, or False if the led is off
+ */
+static inline bool pico_status_led_get(void) {
+#if defined PICO_STATUS_LED_PIN
+    return gpio_get(PICO_STATUS_LED_PIN);
+#elif defined PICO_STATUS_LED_WS2812_PIN
+    return pico_status_led_ws2812_get();
+#elif defined CYW43_WL_GPIO_LED_PIN
+    // Ask the wifi "driver" to set the GPIO on or off
```

Comment:
I guess this comment needs tweaking, given that this is the `_get` function :wink: